fondat/aws/bedrock/resources/sessions.py

```diff
@@ -0,0 +1,289 @@
+"""Resource for managing agent sessions."""
+
+from collections.abc import Iterable
+from typing import Any, Mapping
+from datetime import datetime
+
+from fondat.aws.client import Config
+from fondat.resource import resource
+from fondat.security import Policy
+
+from ..clients import runtime_client
+from ..decorators import operation
+from ..pagination import paginate
+
+
+@resource
+class SessionsResource:
+    """
+    Resource for managing agent sessions.
+    Provides access to create, retrieve, update, and delete sessions.
+    """
+
+    __slots__ = ("_agent_id", "config_runtime", "policies")
+
+    def __init__(
+        self,
+        agent_id: str,
+        *,
+        config_runtime: Config | None = None,
+        policies: Iterable[Policy] | None = None,
+    ):
+        self._agent_id = agent_id
+        self.config_runtime = config_runtime
+        self.policies = policies
+
+    @operation(method="post", policies=lambda self: self.policies)
+    async def create(
+        self,
+        *,
+        encryptionKeyArn: str | None = None,
+        sessionMetadata: dict[str, str] | None = None,
+        tags: dict[str, str] | None = None,
+    ) -> Mapping[str, Any]:
+        """
+        Create a new session.
+
+        Args:
+            encryptionKeyArn: Encryption key ARN
+            sessionMetadata: Session metadata
+            tags: Session tags
+
+        Returns:
+            Session details
+        """
+        params: dict[str, Any] = {}
+        if encryptionKeyArn:
+            params["encryptionKeyArn"] = encryptionKeyArn
+        if sessionMetadata:
+            params["sessionMetadata"] = sessionMetadata
+        if tags:
+            params["tags"] = tags
+        async with runtime_client(self.config_runtime) as client:
+            return await client.create_session(**params)
+
+    @operation(method="get", policies=lambda self: self.policies)
+    async def get(self, sessionIdentifier: str) -> Mapping[str, Any]:
+        """
+        Retrieve details of a specific session.
+
+        Args:
+            sessionIdentifier: Session identifier
+
+        Returns:
+            Session details
+        """
+        async with runtime_client(self.config_runtime) as client:
+            return await client.get_session(sessionIdentifier=sessionIdentifier)
+
+    @operation(method="delete", policies=lambda self: self.policies)
+    async def delete(self, sessionIdentifier: str) -> None:
+        """Delete session.
+
+        Args:
+            sessionIdentifier: Session identifier
+        """
+        async with runtime_client(self.config_runtime) as client:
```

Comment:
We might have to call `end` before we actually delete the session. This might be a ease-of-use wrapper that does a little more for the user, than just returning an error if the session is still active.